PURL-TYPES.rst

```diff
@@ -348,9 +348,12 @@ pypi
 - PyPi treats ``-`` and ``_`` as the same character and is not case sensitive.
   Therefore a Pypi package ``name`` must be lowercased and underscore ``_``
   replaced with a dash ``-``
+- The ``format`` qualifier specifies the type of distribution; one of ``sdist``, ``wheel``, or ``egg``.
```

Comment:
Should consumers assume one of these values if it is present?

Reply:
Now that you mention it, I'm doubting that `format` + `tag` is the right choice.

- I can't think of a situation where you'd want to specify one but not the other (except for sdist, where there is no tag).
- The url is still ambiguous with respect to file extension:
  - Source distributions support zip and tar.gz, and formerly tgz, tar, tar.Z, etc.
  - Binary distributions support egg and exe, possibly others. Example: [zope 4.0.2](https://pypi.org/project/zope.interface/4.0.2/#files), which has both .egg and .exe.

Instead, what about a single qualifier for the full tag + extension suffix? (Could use a better name than `suffix`.) Examples:

- `pkg:pypi/zope.interface@4.0.2?suffix=4.0.2.win-amd64-py2.7.exe`
- `pkg:pypi/django@1.11.1?suffix=py3-none-any.whl`

@di thoughts? As a background, PURL is a universal URL format for all package types, and I'm trying to add a query parameter to narrow scope to a specific artifact for packages on pypi.

Reply:
> Should consumers assume one of these values if it is present?

No, none of these are guaranteed to be present

> Now that you mention it, I'm doubting that format + tag is the right choice.

Why not just use the filename? It's a) guaranteed to be unique and b) contains all possible 'qualifiers' you're adding here and c) does it in a format that's standardized in the Python community.

e.g.
* `zope.interface-4.0.2.win-amd64-py2.7.exe`
* `Django-1.11.1-py2.py3-none-any.whl`

Reply:
[Sorry I let this drop. Picking this up again...]

I was trying to de-duplicate with the package name. In other words, it's not super desirable to say `pkg:pypi/zope.interface@4.0.2?suffix=zope.interface-4.0.2.win-amd64-py2.7.exe`. See how `zope.interface` and `4.0.2` is listed twice? Is there a good way to avoid that?

Reply:
I'm in agreement to use the filename, that would also future-proof the implementation against new wheel platforms/tags.

Reply:
Switched to `filename`.